plugin/completion.py

```diff
@@ -252,7 +261,10 @@ def handle_response(self, response: dict):
             # if insert_best_completion was just ran, undo it before presenting new completions.
             prev_char = self.view.substr(self.view.sel()[0].begin() - 1)
             if prev_char.isspace():
-                self.view.run_command("undo")
+                if last_text_command == "insert_best_completion":
+                    self.view.run_command("undo")
+                else:
+                    return
```

Comment:
This `else: return` has some unintended consequences which I missed before merging :D

Reply:
I put the return there to avoid things like this
<img width="418" alt="screen shot 2017-11-17 at 12 23 30 am" src="https://user-images.githubusercontent.com/1690993/32930648-f23bb3be-cb2d-11e7-96bf-437671397099.png">

It could be reproduced by typing `y` and `<space>` very quickly or when the language server responds slowly.

We might need to check if the context which triggered the completion is still a prefix of the current cursor.

Reply:
Thanks for explaining and sorry I stomped on the changes. I’ll try to integrate it into the existing stale-request handling logic